Darkie changes

TC-DarkSlayer opened the pull request:
Since "4x bullet" bug was supposedly fixed, armor durability was scaled back down again, but its durability is still better than previous update.
Reduced knockback from shotguns by 30% as shotguns shoots a lot more bullets than other guns, causing people to fly in the air due to the sheer knockback.

DARK-DarkSlayer committed:
Armor Durability

- Ballistic Armor durability decrease

DARK-DarkSlayer committed:
Shotgun Knockback

- Reduces knockback from shotguns by 30%

xcs011 commented:
literally when you crackhead
unless it was fixed in actually the last 8 hours it didnt happen

laikadoggirl commented:
also why do you have two separate githubs to make the pr with one and commit with the other wth

TC-DarkSlayer commented:
> also why do you have two separate githubs to make the pr with one and commit with the other wth

Forgot account pass then remembered
Unless you're talking about why I have 3 branches

laikadoggirl commented:
Also @TC-DarkSlayer the knockback change is no longer needed, it was due to the 4x damage glitch as well

Shotguns also shot 4x the number of pellets, and had 4x the knockback

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Equipment/MilitaryHelmet/militaryhelmet_effect.as:
```diff
@@ -58,7 +58,7 @@ void onTick(CBlob@ this)
         headoffset += Vec2f(0, -1);
         milhelmet.SetOffset(headoffset);
-		milhelmet.SetFrameIndex(Maths::Floor(this.get_f32("militaryhelmet_health") / 10.01f));
+		milhelmet.SetFrameIndex(Maths::Floor(this.get_f32("militaryhelmet_health") / 7.51f));
```
indent

Vam-Jam commented:
Holding this back an update (will include in a hotfix). Waiting to see if bullet damage is finally fixed

DARK-DarkSlayer committed:
FleshHit Fix

- A little bit of rearranging to make it nicer when coding in new armor
- Fixes rare divide by 0 bug

DARK-DarkSlayer committed:
Assigning Value "UB" Error

- Assigns value to f32 armorMaxHealth in the case that for some reason the equipment doesn't give it a value

DARK-DarkSlayer committed:
Indent

- Looked fine to me tho, prob some form of hidden space caused weird indent

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -97,12 +97,17 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 		{
-			f32 armorMaxHealth;
-			f32 ratio;
+			f32 armorMaxHealth=10.0f;
```
armorMaxHealth = 10.0f, space for readability

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -141,12 +143,14 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 		{
-			f32 armorMaxHealth;
-			f32 ratio;
+			f32 armorMaxHealth=10.0f;
```
same here

Vam-Jam commented on TFlippy_TerritoryControl_Core_Dev/Entities/FleshHit.as:
```diff
@@ -188,12 +191,11 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
-			f32 ratio;
-
+			f32 armorMaxHealth=10.0f;
+			f32 ratio=0.0f;
```
and here

Vam-Jam commented:
also fix merge conflicts if you can

TC-DarkSlayer commented:
This PR was based on a PR that was reverted
Requires this revert to be reverted 
![image](https://user-images.githubusercontent.com/68712994/126526850-5646276d-418e-4d5a-98c9-9c138bf19bf0.png)

DARK-DarkSlayer committed:
Revert "Revert "Merge pull request #238 from TC-DarkSlayer/DarkieChanges""

This reverts commit 45b1a59ac672d0475b4a4d4911511b507f3f0caf.

DARK-DarkSlayer committed:
Merge branch 'master' into DarkArmor

DARK-DarkSlayer committed:
Readable code

- Making code more legible
- Fix for militaryhelmet wrong health

DARK-DarkSlayer committed:
Organization and Tweaks

- Rearranged/Sorted TinkerTable
- Lighter now costs lantern instead of oil
- Grinder gives more wood from logs

DARK-DarkSlayer committed:
Backwall Coins

- Placing backwalls for wood, stone, concrete, iron, and plasteel now gives coins
- wood/stone = 1 coin
- concrete/iron = 2 coins
- plasteel = 4 coins

DARK-DarkSlayer committed:
Food Changes

- Various types of food items now stack
- Food that are stacked are no longer separate blobs to reduce lag
- Extractor range is now 4 blocks
- Grinded stacked steak is accounted for in grinder

DARK-DarkSlayer committed:
Double Heal Fix

- Possible Fix for double healing (Could be used by vanilla if they wanted to make food stackable)

DARK-DarkSlayer committed:
Chicken Grenade

- UPF PlayerChickens can now activate and throw fraggrenades

DARK-DarkSlayer committed:
Juggernaut Eat

- Juggernauts can now utilize the EatFoodButton

DARK-DarkSlayer committed:
Kudzu Nerf

- KudzuCores are no longer immune to radiation
- KudzuCores no longer reduce dmg from bombs by 75% now increases by 25% (-75% dmg to +25%)
- KudzuCores are no longer super resistant to bullets

DARK-DarkSlayer committed:
Indents

DARK-DarkSlayer committed:
TeamStorage + BluePrints

- Fixed TeamStorage scam bug
(Incorrectly takes more mats than it should from your team storage)
- Blueprints won't be consumed/transported from its original location upon usage.
(blueprint in team storage will stay in team storage even if used remotely)

TC-DarkSlayer reviewed (commented):
- Building Druglab gives 33% chance to give you chemistry blueprint
(may tweak percent chances later on, we'll see for now how this gacha mechanic goes)

DARK-DarkSlayer committed:
TeamStorage and !mats

- !mats command for testers
- Fixed a team storage bug that doesn't count taken mats
(3 citadels each with 90 wood, if I build a buildershop(100 wood) using only team storage, it will consume all 270 wood)

DARK-DarkSlayer committed:
Druglab GachaChemsheet

- Constructing a druglab has a 1/3 chance of giving you a chemsheet blueprint

DARK-DarkSlayer committed:
Armor Durability

- Armor durability slightly reduced
(again)
- bulletproofvest: 35 to 30
- helmet: 30 to 25

TC-DarkSlayer commented:
Current Summary:
	Code Stuff: indents, and making things use switch case to make it faster
	New Armor : Scubagear, minershelmet, pumpkin, combatboots (now protects minor damage)
	Ballistic Armor now has more durability
	Shotgun 30% less knockback
	Repairable Armor and Bushyhelmet(camo) upgrade
	Tinkertable now sorted
	Lighter costs a lantern instead of oil
	Grinder gives a little bit more wood
	Placing backwalls gives coins
	Food is now stacked into one blob, reducing lag
	Juggernaut can now use V to eat
	PlayerChickens can now throw grenades
	Fixed teamstorage bugs
	Fixed double count faction upgrading
	Upgrading/Building/Repairing faction bases now costs less as a result of above
	Armory gives teamStorage remote access
	Making Druglabs has chance of dropping blueprint: Chemistry
	Blueprints will no longer be consumed
	Royal guard can use flippers
	GrainPlants now drop 1-3 grain
	Grain Stacks to 10
	Badgers drop less meat
	Can craft coal from forges
	Food cans costs 2x more
	Reduced bullet resistance of dirt
	Engineer tools can be made via tinkertable
	Foof AoE effect caps at 5 foof
	Domino and crak speed effect reduced
	Dangerous Domino Recipe buffed
	Fumes duration buffed
	Crak no longer has damage reduction
	KudzuRework:
		KudzuCores can now get specific mutations by feeding the following:
		mithril ingot/ore/enriched, dirt, oil, meat, steak, goldore, badger, and sulphur
		Mithril ingots are used to hasten the mutation rates
		KudzuCores no longer has damage reduction from bullets/bombs
		KudzuCores can now be damaged by radiation
		Kudzu now can mutate radiation resistance
		KuzuCores costs 5 mithril ingots
		Kudzu no longer has damage reduction from bullets/bombs
		Kudzu can now be damaged by radiation

Wunarg commented:
Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

DARK-DarkSlayer committed:
Kudzu Flammable

- Kudzu tiles are now flammable
- Some code cleanup/ changing to switch to improve code...?

TC-DarkSlayer commented:
> Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

Was pain in a neck to destroy kudzu cores, took me around 8 dynamite per kudzucore(kudzu tiles kept regening) vs mason's 15 kudzu cores

Wunarg commented:
> 
> 
> > Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else
> 
> Was pain in a neck to destroy kudzu cores, took me around 8 dynamite per kudzucore(kudzu tiles kept regening) vs mason's 15 kudzu cores

thats what i mean though, your not supposed to use explosives against kudzu

DARK-DarkSlayer committed:
Automat Targetting

- Automats/Fishturrets no longer target fish, chickens, landfish, and pigs
- royalguards can now use flippers
(but the gained swimspeed is still less than others)

DARK-DarkSlayer committed:
Kudzu Rework

- Kudzu can now be fed a variety of things:
   - Mithrilingot/ore/enriched, dirt, sulphur, oil, gold ore, meat, steak, badgers
- Kudzu can now mutate mithril resistance
- Kudzu's mutation time depends on additional mithril ingots fed

DARK-DarkSlayer committed:
Faction Upgrade Fix

- Upgrading faction bases no longer double counts
- Lowering costs of upgrading/repairing faction bases to combat this
- ChickenBase Health lowered
- Armory has team storage
- GrainPlants now drop 1-3 grain

DARK-DarkSlayer committed:
Kudzu Cost

- Kudzu Cores now costs 5 mithril ingots

DARK-DarkSlayer committed:
FactionBase Lights

- Faction Bases now emit light
- Faction Alarm now properly turns off
- Reduced volume of alarm by 25%

DARK-DarkSlayer committed:
Small Changes

- Grain Stacks to 10
- Badgers drop less meat
- Can craft coal from forges
- Food cans costs 2x more
- Reduced bullet resistance of dirt
- Engineer tools can be made via tinkertable

DARK-DarkSlayer committed:
Drug Changes

- Foof now has a cap on its AoE effect
- Domino speed effect nerf
- Dangerous Domino recipe buffed
- Crak speed effect nerf
- Fumes duration buffed
- Fixed domino bug that set_f32 instead of add_f32 (every other drug had this)
- Fixed entire stack of grain consumed by druglab (since food now stacks into one blob if condition is so that you cant get 0 grainstack in inventory do !grain 0 and you'll see)

TC-DarkSlayer committed:
Merge branch 'master' into DarkieChanges

TC-DarkSlayer committed:
ResolveConflicts

- "RadResistance" to "RadiationResistance"

laikadoggirl commented:
I don't think shotgun needs a further knockback nerf, as that was majorly fixed when the 4x bullets bug was.
It would also almost make more sense to just make the chemlab have a chance to drop the blueprint when created, like you got experience from building it, and made the blueprint

laikadoggirl commented:
Reducing dirt bullet resistance is eh too, rather I think it should be like neutrals with concrete, where they can damage it, but never fully break it

charliehstreet commented:
neuts can completely break concrete, you're thinking of reinforced steel concrete. Also autochemlab should drop its blueprint everytime its destroyed, along with the missile assembler and the sun tower

laikadoggirl commented:
Oh, I literally meant that the normal chem lab should have a chance to drop it on construction, albeit a small chance, and yeah I was thinking about reinforced

laikadoggirl commented:
That or ill finally add a sandbag block to absorb bullets

laikadoggirl commented:
> 
> 
> Really not a fan of the kudzu damage resistances changes, since defeating kudzu with bombs and bullets removes one of the intersting aspects of kudzu of needing to use other weaponary, instead of the weaponary used everywhere else

I mean yeah but why wouldnt a plant be just as vulnerable to explosives

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as:
```diff
@@ -71,23 +95,43 @@ void onCommand(CBlob@ this, u8 cmd, CBitStream @params) //Mutate command
+					case -123101143: 	matReq = 100;	break;	// meat
+					case -1370030172:	matReq = 250;	break;	// gold ore
+					case -617913447:	matReq = 250;	break;	// sulphur
+					case 389592510:		matReq = 1;		break;	// badger
```
indent

Vam-Jam commented on TFlippy_TerritoryControl_Misc_Dev/Entities/Misc/Kudzu/KudzuCore.as:
```diff
@@ -54,12 +59,31 @@ const u32 RECHARGETIME = 18000; // 30 = 1 second, 1800 = 1 minute, this is 10 mi
+			case 336243301:		// steak
+				canMutate = true;
+				break;
+		}
```
space

Vam-Jam commented:
This PR has gotten too big, I'll sit down and review it when I get the time.

DARK-DarkSlayer committed:
Acid/Fire Gasses

- Acid now doesn't fully destroy dirt instead of not damaging it at all
- Fire gas now breaks castle blocks

TC-DarkSlayer committed:
Resolve Indents

DARK-DarkSlayer committed:
Merge branch 'DarkieChanges' of https://github.com/TC-DarkSlayer/kag_territorycontrol into DarkArmor

DARK-DarkSlayer committed:
Crak DmgReduct Nerf

- Crak no longer has damage reduction

TC-DarkSlayer committed:
Kudzu no longer isDestroyedFromFire

- Removed flammable tag for damaged kudzu

TC-DarkSlayer commented:
Splitting this PR
